tools/crm_simulate.c

```diff
@@ -702,7 +700,8 @@ profile_one(const char *xml_file, long long repeat, pe_working_set_t *data_set,
         pcmk__schedule_actions(data_set, input, NULL);
         pe_reset_working_set(data_set);
     }
-    printf(" %.2f secs\n", (clock() - start) / (float) CLOCKS_PER_SEC);
+
+    out->message(out, "profile", xml_file, start, clock());
```

Comment:
We now have an easy way of checking what the overhead of formatted output is, using pcmk_simtimes to compare before and after this PR

Reply:
```
$ tools/pcmk_simtimes /tmp/crm_simulate.before /tmp/crm_simulate.after
...
Averages: +0.01s  +2.03%
```

The bigger concerns I have are:

* Tons of this in /tmp/crm_simulate.after:  `Only 'private' parameters to stonith-fence_compute-fence-nova for unfencing compute-1 changed`
* Output to the screen seems to be batched up - it prints some, and then seems to hang for a long time before the next big chunk gets printed.

Reply:
Fixed the buffering.  Still have the harder problem to figure out.

Reply:
All that output is coming from https://github.com/ClusterLabs/pacemaker/pull/2322/commits/7d4c60091d69059597a83dcbe9e353423d467c72#diff-7b3f0ed3ea1b5a120ccc8da18b969eca9e1513b20db3d4665bb23e0efd02f2a7L283.  This didn't show up in crm_simulate profile output in the past because pe_flag_stdout was set after the profile function had run (and then the program exited).

So I'm not really sure how to fix this up so it only happens from crm_simulate sometimes.  We can't just use out->info or it'll always output (and get logged, ugh).  I really don't want to test fmt_name more than I already am, and I'm not sure if that would help in the first place.  Maybe pe_flag_stdout is more useful than we want to admit?

Reply:
I've updated 5e9de5b in a way that fixes this output problem, and I think it's a reasonable approach.